src/components/Blog/BlogSidebarPosts.jsx

```diff
@@ -26,7 +26,7 @@ const BlogSidebarPosts = () => {
 
   useEffect(() => {
     let mounted = true;
-    if (mounted) dispatch(getSidebarPosts());
+    if (mounted && Boolean(!posts.length)) dispatch(getSidebarPosts());

```

Comment:
Co robi `Boolean(!posts.length)`?

W takim wypadku `posts` musisz wrzucić do zależności `useEffect`.

Reply:
to są posty na sidebarze, tam mi wystarczy żeby tylko raz były pobrane, bo to są zawsze 4 najnowsze posty. Dlatego ten warunek mi sprawdza czy one są, bo jeżeli tak to nie ma po co kolejny raz strzelać do API. Jeżeli dodam `posts` do zależności `useEffect` redux mi 3 razy tą samą metodę wywoła.